core/src/main/java/org/apache/accumulo/core/clientImpl/InstanceOperationsImpl.java

```diff
@@ -520,43 +520,64 @@ public ServerId getServer(ServerId.Type type, String resourceGroup, String host,
 
   @Override
   public Set<ServerId> getServers(ServerId.Type type) {
-    return getServers(type, null);
+    AddressPredicate addressPredicate = addr -> true;
+    return getServers(type, rg -> true, addressPredicate);
   }
 
   @Override
-  public Set<ServerId> getServers(ServerId.Type type, Predicate<ServerId> test) {
+  public Set<ServerId> getServers(ServerId.Type type, Predicate<String> resourceGroupPredicate,
+      BiPredicate<String,Integer> hostPortPredicate) {
+    Objects.requireNonNull(type, "Server type was null");
+    Objects.requireNonNull(resourceGroupPredicate, "Resource group predicate was null");
+    Objects.requireNonNull(hostPortPredicate, "Host port predicate was null");
+
+    AddressPredicate addressPredicate = addr -> {
+      var hp = HostAndPort.fromString(addr);
+      return hostPortPredicate.test(hp.getHost(), hp.getPort());
+    };
+
+    return getServers(type, resourceGroupPredicate, addressPredicate);
+  }
+
+  private Set<ServerId> getServers(ServerId.Type type, Predicate<String> resourceGroupPredicate,
+      AddressPredicate addressPredicate) {
+
     final Set<ServerId> results = new HashSet<>();
+
     switch (type) {
       case COMPACTOR:
-        context.getServerPaths().getCompactor(rg -> true, addr -> true, true)
+        context.getServerPaths().getCompactor(resourceGroupPredicate::test, addressPredicate, true)
             .forEach(c -> results.add(createServerId(type, c)));
         break;
       case MANAGER:
         ServiceLockPath m = context.getServerPaths().getManager(true);
-        Optional<ServiceLockData> sld = context.getZooCache().getLockData(m);
-        String location = null;
-        if (sld.isPresent()) {
-          location = sld.orElseThrow().getAddressString(ThriftService.MANAGER);
-          HostAndPort hp = HostAndPort.fromString(location);
-          results.add(new ServerId(type, Constants.DEFAULT_RESOURCE_GROUP_NAME, hp.getHost(),
-              hp.getPort()));
+        if (m != null) {
```

Comment:
This line is fixing a bug I am seeing when running test. When m is null on the next line an NPE will happen w/o this check

Reply:
Could be the Manager is not fully up yet. Probably safe to leave it.